### PR TITLE
Speed up Import(FieldInfo) when debugging

### PR DESCRIPTION
**Pull request by CreateAndInject**
I already reported this issue to Microsoft, see [here](https://github.com/dotnet/runtime/issues/92856)

### CODE REVIEW

**Comment by CreateAndInject** [comment]
Seems this issue only happen when **_unchecked_** `Enable Just My code`

File: src/DotNet/Importer.cs [hunk 1]
```diff
@@ -650,7 +650,7 @@ public Importer(ModuleDef module, ImporterOptions options, GenericParamContext g
 			try {
 				// Get the original field def in case the declaring type is a generic
 				// type instance and the field uses a generic type parameter.
-				origField = fieldInfo.Module.ResolveField(fieldInfo.MetadataToken);
+				origField = fieldInfo.DeclaringType is { IsGenericType: true } ? fieldInfo.Module.ResolveField(fieldInfo.MetadataToken) : fieldInfo;
```
Comment (wtfsck) [inline src/DotNet/Importer.cs:653]: Is this code exactly identical to the original code?

Comment (CreateAndInject) [reply src/DotNet/Importer.cs:653]: I think so, `typeof(C1<>.C2).IsGenericType` is `true`, so if `IsGenericType` is `false`, I don't think `ResolveField` can return a different value
```C#
public class C1<T>
{
    public class C2
    {
        public T Field;
    }
}
```

Comment (wtfsck) [reply src/DotNet/Importer.cs:653]: I can only accept this PR if it's guaranteed to work the same as before.

**Pull request closed without merging**